### PR TITLE
ARC: boards: ARCv3: add qemu_arc_hs5x board

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
Add qemu_arc_hs5x - single core ARCv3 32bit HS5x CPU.

NOTE: we don't enable zephyr toolchain for it as current Zephyr SDK doesn't have hs5x support. We'll enable it when we switch to 0.15.0 Zephyr SDK.

While I'm at it also do minor non-functional polishing for HS6x QEMU and update board documentation.

### CODE REVIEW

**Review by abrodkin — DISMISSED** [review]

**Review by ruuddw — COMMENTED** [review]
See minor comments. Also, any documentation to be updated with this new board?
ARC support matrix has HS 5x already included, but not 6x. ARC QEMU board documentation I think is generic, nothing specific for 3x/4x/5x/6x, right?

**Comment by evgeniy-paltsev** [comment]
Hi @ruuddw,

> ARC support matrix has HS 5x already included, but not 6x. 

Hmm, we have both hs6x and hs5x in the latest docs:
https://docs.zephyrproject.org/latest/hardware/arch/arc-support-status.html#support-status
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/hardware/arch/arc-support-status.rst#support-status

Though in the doc page there is horizontal scroll for the table - looks like it isn't really noticeable :/

File: boards/arc/qemu_arc/qemu_arc_hs5x.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+identifier: qemu_arc_hs5x
+name: QEMU Emulation for ARC HS
```
[the remark this answers is not in the document]
Comment (evgeniy-paltsev) [reply boards/arc/qemu_arc/qemu_arc_hs5x.yaml]: done, thanks!

File: soc/arc/snps_qemu/Kconfig [hunk 2]
```diff
@@ -15,4 +15,7 @@ config SOC_QEMU_ARC_HS
 config SOC_QEMU_ARC_HS6X
 	bool "Synopsys ARC HS in QEMU"
 
+config SOC_QEMU_ARC_HS5X
+	bool "Synopsys ARC HS5x in QEMU"
```
[the remark this answers is not in the document]
Comment (evgeniy-paltsev) [reply soc/arc/snps_qemu/Kconfig:19]: done, thanks!

**Review by ruuddw — APPROVED** [review]
Thanks!

**Review by abrodkin — APPROVED** [review]

**Pull request merged**